meeteval/io/seglst.py

```diff
@@ -576,47 +584,71 @@ def apply_multi_file(
-                f'Drop them.',
+                f'{len(h_minus_r)} of {len(hypothesis)} keys are present in '
+                f'the hypothesis but missing in the reference. \n'
+                f'Missing (showing first 5): {h_minus_r[:5]}\n'
+                f'The computation continues with the assumption that only the '
+                f'sub-set of sessions present in the reference should be '
+                f'evaluated.',
             )
             hypothesis = {
                 k: v
                 for k, v in hypothesis.items()
                 if k not in h_minus_r
             }
 
+        # The following if statement is active when keys are missing in the
+        # hypothesis. We assume that the system didn't produce any output for
+        # the missing examples but throw an exception when a threshold is
+        # exceeded
         if len(r_minus_h) == 0:
             pass
         elif ratio <= allowed_empty_examples_ratio:
             logging.warning(
-                f'Missing {ratio * 100:.3} % = '
-                f'{len(r_minus_h)}/{len(reference.keys())} of recordings in'
-                f' hypothesis.\n'
-                f'Please check your system, if it ignored some recordings or '
-                f'predicted no transcriptions for some recordings.\n'
-                f'Continue with the assumption, that the system predicted '
-                f'silence for the missing recordings.',
+                f'Missing {ratio * 100:.3f} % = '
+                f'{len(r_minus_h)}/{len(reference.keys())} of recordings in '
+                f'hypothesis. An exception will be raised when this number '
+                f'exceeds {allowed_empty_examples_ratio * 100:3} %.\n'
+                f'Please check if your system ignored recordings or '
+                f'predicted silence (no transcripts) for some recordings. '
+                f'It is recommended to output an empty transcript if silence '
+                f'was recognized.\n'
+                f'Missing (showing first 5): {r_minus_h[:5]}\n'
+                f'The computation continues with the assumption that the system '
+                f'predicted silence for the missing recordings.',
             )
         else:
             raise RuntimeError(
-                'Keys of reference and hypothesis differ\n'
-                f'hypothesis - reference: e.g. {h_minus_r[:5]} '
-                f'(Total: {len(h_minus_r)} of {len(hypothesis)})\n'
-                f'reference - hypothesis: e.g. {r_minus_h[:5]} '
-                f'(Total: {len(r_minus_h)} of {len(reference)})'
+                f'Missing {ratio * 100:.3f} % = '
+                f'{len(r_minus_h)}/{len(reference.keys())} of recordings in '
+                f'hypothesis. This exceeds the threshold of '
+                f'{allowed_empty_examples_ratio * 100:3} %.\n'
+                f'Please check if your system ignored recordings or '
+                f'predicted silence (no transcripts) for some recordings. '
```

Comment:
Is empty transcript fine?

`Please check if your system ignored recordings or predicted nothing (no transcripts) for some recordings`

My interpretation: `silence` -> `empty transcript`

Reply:
I removed the second part of the sentence. I think that makes it easier to understand